### ISSUE TITLE
Can't remove DRM on azw3 ebook. IndexError raised.

### ISSUE THREAD

**Issue by Lutalo** [noDRM/DeDRM_tools#17]
Hi all, I've probably missed something fairly obvious here while setting this up.

I downloaded the azw3 file from Amazon with the "Download & Transfer via USB" in the "Manage digital content" section of my account. I also have added my Kindle Paperwhite's serial number inside the configuration for DeDRM.

I also tried downloading the ebook with Kindle for PC 1.17 and the same error occurs.

It seems to be throwing an IndexError instead of one of the other exceptions I see others encountering in their issues.
`IndexError: Replacement index 1 out of range for positional args tuple`

**OS**: Windows 10
**Calibre version**: 5.34
**DeDRM version**: 10.0.2

Debug log output
```
calibre Debug log
calibre 5.34 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.19041 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 46, 0) && DeDRM (10, 0, 2) && Package KFX (from KFX Input) (1, 46, 0) && KFX metadata reader (from KFX Input) (1, 46, 0) && KFX Input (1, 46, 0) && Obok DeDRM (10, 0, 0)
calibre 5.34 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.19041 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 46, 0) && DeDRM (10, 0, 2) && Package KFX (from KFX Input) (1, 46, 0) && KFX metadata reader (from KFX Input) (1, 46, 0) && KFX Input (1, 46, 0) && Obok DeDRM (10, 0, 0)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 95.6235294117647 x 94.59310344827585
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.47] splash screen shown
[0.47] Initializing db...
[0.47] db initialized
[0.47] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
[0.95] main UI initialized...
[0.95] Hiding splash screen
Starting QuickView
DeDRM v10.0.2: Trying to decrypt Assistive Technologies- E-Book.azw3
Using Library AlfCrypto DLL/DYLIB/SO
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v1.0.
Copyright © 2008-2020 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Assistive Technologies- E-Book
Found 0 keys to try after 0.1 seconds
Crypto Type is: 2
DeDRM v10.0.2: Failed to decrypt with error: Replacement index 1 out of range for positional args tuple
DeDRM v10.0.2: Looking for new default Kindle Key after 0.1 seconds
searching for kinfoFiles in C:\Users\Steven\AppData\Local
Found K4PC 1.9+ kinf2011 file: C:\Users\Steven\AppData\Local\Amazon\Kindle\storage\.kinf2011
Decrypted key file using IDString '3231205406' and UserName 'Steven'
DeDRM v10.0.2: Found 1 new key
MobiDeDrm v1.0.
Copyright © 2008-2020 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Assistive Technologies- E-Book
Got DSN key from database default_key_1
Found 4 keys to try after 0.2 seconds
Crypto Type is: 2
DeDRM v10.0.2: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 795, in KindleMobiDecrypt
  File "calibre_plugins.dedrm.k4mobidedrm", line 244, in GetDecryptedBook
  File "calibre_plugins.dedrm.mobidedrm", line 464, in processBook
IndexError: Replacement index 1 out of range for positional args tuple

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "calibre\customize\ui.py", line 175, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 888, in run
  File "calibre_plugins.dedrm.__init__", line 838, in KindleMobiDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v10.0.2: Ultimately failed to decrypt after 0.2 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
[5.41] splash screen hidden
[5.41] Started up in 5.41 seconds with 3 books
stdout+stderr from file dialog helper: [b'', b'']
piped data from file dialog helper: [b'\x12\xd0\x93\xc9\x1bW\xf7 j\x860\\\xc9)>1`\x95h\x85\xf6g;\x1f\x8d\x95;\x04\xec\xf0\xdd\xe5', b'C:\\Users\\Steven\\Downloads\\Assistive Technologies- E-Book.azw3']
Added Assistive Technologies- E-Book to db in: 0.1
Added 1 books in 0.6 seconds
```

Thanks for taking the time to look over all this!

**Comment by ElleKayEm** [comment]
It looks like you didn't have your Kindle's serial number in there when you made this log:

> Found 0 keys to try after 0.1 seconds

**Comment by noDRM** [comment]
Not sure about the serial number issue, but that would lead to a different error message. The IndexError is now fixed, you can download the most recent version by clicking on the "Actions" tab, then on the first (latest) entry, then the "plugin" file under "Artifacts". 

Give that a try and see if that works. If it doesn't, make sure you have the serial number entered under "Kindle eInk ebooks" in the settings and post another log please.

**Comment by Lutalo** [comment]
Looks like @ElleKayEm was right, after applying the fix in 4d43685 I now end up with no keys found. I guess I'm not sure how to enter the Kindle serial number correctly? I'm getting the serial number from the Kindle's **Settings** -> **Device Info** -> **Serial Number**. 

HOWEVER, importing the book with the azw file I download through Kindle For PC 1.17 DOES work 🥳. I'm able to successfully import and convert the book.
After this successfully imported, I went back and checked my serial number settings and it turns out Calibre didn't save them. After _really_ making sure the serial number is added, I imported the azw3 file again and it works! 

I'm attaching the debug log from my failed Kindle azw3 import just for completeness sake, but it looks to have been user error on my part. Thanks all!

```calibre Debug log
calibre 5.34 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.19041 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 46, 0) && DeDRM (10, 0, 2) && Package KFX (from KFX Input) (1, 46, 0) && KFX metadata reader (from KFX Input) (1, 46, 0) && KFX Input (1, 46, 0) && Obok DeDRM (10, 0, 0)
calibre 5.34 [64bit]  embedded-python: True is64bit: True
Windows-10-10.0.19041 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19041')
Python 3.8.5
Windows: ('10', '10.0.19041', '', 'Multiprocessor Free')
Interface language: None
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 46, 0) && DeDRM (10, 0, 2) && Package KFX (from KFX Input) (1, 46, 0) && KFX metadata reader (from KFX Input) (1, 46, 0) && KFX Input (1, 46, 0) && Obok DeDRM (10, 0, 0)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 95.6235294117647 x 94.59310344827585
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.47] splash screen shown
[0.47] Initializing db...
[0.47] db initialized
[0.47] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
[0.95] main UI initialized...
[0.95] Hiding splash screen
Starting QuickView
DeDRM v10.0.2: Trying to decrypt Assistive Technologies- E-Book.azw3
Using Library AlfCrypto DLL/DYLIB/SO
Using Library AlfCrypto DLL/DYLIB/SO
MobiDeDrm v1.0.
Copyright © 2008-2020 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Assistive Technologies- E-Book
Found 0 keys to try after 0.2 seconds
Crypto Type is: 2
Warning: This is a library or rented ebook (1643889599000). Continuing ...
DeDRM v10.0.2: Failed to decrypt with error: No key found in 0 PIDs tried.
DeDRM v10.0.2: Looking for new default Kindle Key after 0.2 seconds
searching for kinfoFiles in C:\Users\Steven\AppData\Local
Found K4PC 1.9+ kinf2011 file: C:\Users\Steven\AppData\Local\Amazon\Kindle\storage\.kinf2011
Decrypted key file using IDString '3231205406' and UserName 'Steven'
DeDRM v10.0.2: Found 1 new key
MobiDeDrm v1.0.
Copyright © 2008-2020 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: Assistive Technologies- E-Book
Got DSN key from database default_key_1
Found 4 keys to try after 0.3 seconds
Crypto Type is: 2
Warning: This is a library or rented ebook (1643889599000). Continuing ...
DeDRM v10.0.2: Ultimately failed to decrypt after 0.3 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 795, in KindleMobiDecrypt
  File "calibre_plugins.dedrm.k4mobidedrm", line 244, in GetDecryptedBook
  File "calibre_plugins.dedrm.mobidedrm", line 500, in processBook
calibre_plugins.dedrm.mobidedrm.DrmException: No key found in 0 PIDs tried.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "calibre\customize\ui.py", line 175, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 888, in run
  File "calibre_plugins.dedrm.__init__", line 838, in KindleMobiDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v10.0.2: Ultimately failed to decrypt after 0.3 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
[146.55] splash screen hidden
[146.55] Started up in 146.55 seconds with 3 books
stdout+stderr from file dialog helper: [b'', b'']
piped data from file dialog helper: [b'n\xffg\xf1\xcb\x07t\x159\x9f\x8c\xc8x8\x9bhMCa\x17\x92h\x16804\xcb)\xdd\xe7\x86\x06', b'C:\\Users\\Steven\\Downloads\\Assistive Technologies- E-Book.azw3']
Added Assistive Technologies- E-Book to db in: 0.1
Added 1 books in 0.7 seconds
```
